app/src/renderer/vuex/modules/send.js

```diff
@@ -97,18 +118,30 @@ export default ({ node }) => {
       delete args.to
       delete args.pathParameter
 
-      // get the generated tx by querying it from the backend
-      let request
-      if (to && pathParameter) {
-        request = node[type](to, pathParameter, args)
-      } else if (to) {
-        request = node[type](to, args)
-      } else {
-        request = node[type](args)
-      }
+      // simulation to get the estimated gas
+      let request = actions.apiRequest(type, to, pathParameter, args)
+      const simulationRes = await request.catch(handleSDKError)
+      const adjustedGas = Number(simulationRes.gas_estimate) * gasAdjustment
+      const estimatedFeesLow =
+        Number(simulationRes.gas_estimate) * Number(gasPrices[0].amount) * 1e-7
+      const estimatedFeesHigh =
+        Number(adjustedGas) * Number(gasPrices[0].amount) * 1e-7
+
+      console.log(
```

Comment:
console.log? is this temporary until the info is displayed in the UI?

Reply:
yup, as @faboweb suggested

Reply:
I can work on the action modal as soon as we merge this